src/Http/HttpRequester.php

```diff
@@ -0,0 +1,187 @@
+<?php
+
+namespace Castor\Http;
+
+use Castor\Helper\PathHelper;
+use Psr\Log\LoggerInterface;
+use Psr\Log\NullLogger;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+
+/** @internal */
+class HttpRequester
+{
+    public function __construct(
+        private readonly HttpClientInterface $httpClient,
+        private readonly Filesystem $filesystem,
+        private readonly LoggerInterface $logger = new NullLogger(),
+    ) {
+    }
+
+    public function httpClient(): HttpClientInterface
+    {
+        return $this->httpClient;
+    }
```

Comment:
There is no need for this public method. Could you remove it?

Reply:
I've removed it but let's keep in mind that it's used in `GlobalHelper::getHttpClient()`, a non internal class that userland could have relied on. So a BC BREAK in theory.